### ISSUE TITLE
Add test for #1372 (short-lived client)

### ISSUE THREAD

**Issue by blueyed** [awesomeWM/awesome#1443]
https://github.com/awesomeWM/awesome/pull/1372

**Comment by psychon** [comment]
I don't think it is easy to add a test for that. The bug you are testing for is "awesome manages a new client, but in a following delayed_call the client is already gone". This means that managing and unmanaging have to happen in the same main loop iteration. This means that the client cannot wait until it is managed and then close quickly, because this is (likely) already too late.

However, if a new client opens and closes immediately, you are now racing this code: https://github.com/awesomeWM/awesome/blob/master/event.c#L778-L783
This is: We get a map request (a new window is opened), but asking for its geometry fails. This would mean the X11 window is already destroyed at this point.

Thus, you need to MapWindow and immediately UnmapWindow (X11-speak here). I don't think that this is something that Gtk can provide. I would expect it to DestroyWindow instead of UnmapWindow.

Just wanted to warn you...
